### DISCUSSION TITLE
Persisting state in the browser

### DISCUSSION THREAD

**Discussion by tclasen** [idom-team/idom#662] [Question] [answered]
When running a stateful web app (incrementing counter) I noticed that if my server hot-reloads I lose all state in my browser. Is it possible to shift some of the state to the frontend to allow for load balanced servers where you can't always be on the same server each request? Or does idom rely on the web sockets for state transfer even in production mode?

**Comment by Archmonger** [comment, marked as answer]
In order for anything to be persistent, you'll need some sort of persistent storage, such as a database.

If your web framework of choice (FastAPI, Flask, etc) does not have integrated database capabilities, you might want to pip install an ORM such as [SQLAlchemy](https://pypi.org/project/SQLAlchemy/).

**Reply by tclasen** [reply]
Sorry, I'm talking about a purely client-side persistence in the DOM. Is the vdom for a given client connection stored client-side or server-side?

**Reply by Archmonger** [reply]
`IDOM` is strictly reliant on websockets, and as of right now, neither myself nor Ryan have though of a way to securely maintain pure client-side persistence without adding security vulnerabilities. Especially in a situation of distributed servers.

Unless one day we determine a solution, any persistence of values needs to be handled within the backend. I'd recommend setting your default values within `idom.hooks.use_state` to be fetched from a database. Then, synchronize your database values when appropriate.

### Relevant Links
- https://github.com/idom-team/django-idom/issues/34
- https://github.com/idom-team/django-idom/discussions/19
